examples/plot_variable_importance_classification.py

```diff
@@ -0,0 +1,156 @@
+"""
+Variable Importance for Classification
+=============================================================
+"""
+
+# %%
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from joblib import Parallel, delayed
+from scipy.stats import ttest_1samp
+from sklearn.base import clone
+from sklearn.datasets import load_wine
+from sklearn.linear_model import LogisticRegressionCV, RidgeCV
+from sklearn.metrics import balanced_accuracy_score, hinge_loss, log_loss
+from sklearn.model_selection import KFold
+from sklearn.svm import SVC
+
+from hidimstat import CPI, LOCO, PermutationImportance, desparsified_lasso
+
+# %%
+#############################################################################
+# Generate data where classes are not linearly separable
+# --------------------------------------------------------------
+rng = np.random.RandomState(0)
+dataset = load_wine()
+X, y = dataset.data, dataset.target
+labels = dataset.feature_names
+labels
+# %%
+cv = KFold(n_splits=5, shuffle=True, random_state=0)
+
+df_list = []
+model_linear = LogisticRegressionCV(Cs=np.logspace(-3, 3, 10))
+model_svc = SVC(kernel="rbf", random_state=0)
+
+
+def run_one_fold(X, y, model, train_index, test_index):
+
+    X_train, X_test = X[train_index], X[test_index]
+    y_train, y_test = y[train_index], y[test_index]
+
+    model_ = clone(model)
+    model_.fit(X_train, y_train)
+    y_pred = model_.predict(X_test)
+
+    return balanced_accuracy_score(y_test, y_pred)
+
+
+scores = Parallel(n_jobs=5)(
+    delayed(run_one_fold)(X, y, model_linear, train_index, test_index)
+    for train_index, test_index in cv.split(X)
+)
+scores
+# %%
+
+
+for train_index, test_index in cv.split(X):
+    X_train, X_test = X[train_index], X[test_index]
+    y_train, y_test = y[train_index], y[test_index]
+
+    model_linear_ = clone(model_linear)
+    model_linear_.fit(X_train, y_train)
+    model_svc_ = clone(model_svc)
+    model_svc_.fit(X_train, y_train)
+
+    cpi_linear = CPI(
```

Comment:
you should create the object before the loop with a short description of the method.

Reply:
The class's `__init__` takes a fitted model as an argument. It cannot be created before fitting the model, which is done in the loop.

Reply:
Good point.

Reply:
Good point.

Reply:
In my opinion, it's better to create multiple loops when you create an example and do a small number of calculi than a big loop with everything.

You can generate all the estimators before to instantiate CPI.

Reply:
Also, note that the code is encapsulated in a function. The idea is to show that the code is easy to parallelize.